readthedocs/projects/forms.py

```diff
@@ -1025,8 +1026,35 @@ def clean_domain(self):
             if invalid_domain and domain_string.endswith(invalid_domain):
                 raise forms.ValidationError(f"{invalid_domain} is not a valid domain.")
 
+        if self._has_cname(domain_string):
+            raise forms.ValidationError(
+                _(
+                    "This domain has a CNAME record. "
+                    "Please remove it before adding the domain.",
+                ),
+            )
+
         return domain_string
 
+    def _has_cname(self, domain):
+        """
+        Check if a domain has a CNAME record.
+
+        We make use of `dig` to check if a domain has a CNAME record,
+        if it doesn't return an empty string, it means it has a CNAME record.
+        """
+        try:
+            result = subprocess.run(
+                ["dig", "-t", "CNAME", "+short", domain],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                timeout=1,
+                check=True,
+            )
+        except (subprocess.TimeoutExpired, subprocess.CalledProcessError):
+            return False
```

Comment:
We could also error on the safe side, and return True here, or maybe another error `error connecting to domain` or something

Reply:
> maybe another error `error connecting to domain` or something

If it's not a lot more work, I think we should do this. It will be more explicit to users for this case.
I think we can raise the exception from here direction, right?

```python
         raise forms.ValidationError(
                _(
                    "There was a connection error. "
                    "Please, try to add the domain in a few minutes.",
                ),
            )
```